lib/waterline/query/deferred.js

```diff
@@ -271,6 +273,22 @@ Deferred.prototype.populate = function(keyName, criteria) {
 };
 
 /**
+ * Add projections to the parent
+ *
+ * @param {Array} attributes to select
+ * @return this
+ */
+
+Deferred.prototype.select = function(attributes) {
+  if(!_.isArray(attributes)) {
+    attributes = [attributes];
+  }
+
+  this._criteria.select = attributes;
```

Comment:
It should append to existing select and not replace it. Will allow to call select twice.